### PR TITLE
Added Find All and zoom button to find and replace panel

### PR DESCRIPTION
**Pull request by Guilty-Spark-343**
### Prerequisites

- [X ] Reviewed the [checklist](CHECKLIST.md)

- [ ] Reviewed feedback from the "Sonar Cloud" bot. Note that you have to wait
    for the "CI / Unit Tests") to complete first. Failed Unit tests can be 
    debugged by adding the label "verbose logging" to the GitHub PR.

### Description of the Change

A new 'Find and Zoom' button added to the Basic Find dialog to zoom to find the matching nodes and to zoom to the selected nodes as noted in #1912.
The action is handled by the zoom to selection plugin.  
find_and_zoom

### Alternate Designs

None considered.

### Why Should This Be In Core?

This find and zoom functionality seems to be a common workflow for users and this PR makes that workflow a little easier. 

### Benefits

Simplified user interaction with the find functionality.

### Possible Drawbacks

Additional button has been added and may cluster the GUI. 

### Verification Process

1. Open the Basic Find dialog in Constellation (Ctrl + f)
2. Enter a search string in the Find field.
3. Click 'Find All and Zoom'
4. Verify the graph view has zoomed into the nodes and edges that match the search string.

### Applicable Issues

#1912

### CODE REVIEW

**Review by antares1470 — CHANGES_REQUESTED** [review]
1) If you could add an entry to whatsnew.txt, that would be great.
2) If you set the search to all open graphs, it only zooms in on the active graph at the time of finding
findViewZoomAllGraphsIssue

**Comment by Guilty-Spark-343** [comment]
Fixed changes, ready to go if there are no other changes.

File: CoreWhatsNewView/src/au/gov/asd/tac/constellation/views/whatsnew/whatsnew.txt [hunk 1]
```diff
@@ -1,6 +1,10 @@
 == 3030-12-31 Getting Started
 <p>If you're new to Constellation, read the <a href="" helpId="au.gov.asd.tac.constellation.functionality.gettingstarted">getting started guide</a>.</p>
 
+
+== 2024-01-08 Added Find and Zoom to BasicFind
+<p> Added a "Find All and Zoom" button to the BasicFind dialog</p>
```
Comment (antares1470) [inline CoreWhatsNewView/src/au/gov/asd/tac/constellation/views/whatsnew/whatsnew.txt:6]: Can you add a space in BasicFind

**Review by antares1470 — CHANGES_REQUESTED** [review]
I'm still encountering zoom only happening on one graph. This time though its only happening on the graph that is open after it has finished searching (which isn't always the graph that was open when the search was submitted

**Comment by Guilty-Spark-343** [comment]
For some reason I wrongly thought the Find All on multiple graphs just returned the first instance.  I've fixed that now, it should zoom on every graph that had results.

**Review by antares1470 — APPROVED** [review]
Once merge conflict is resolved, this should be good now

**Review by Auriga2 — COMMENTED** [review]
Looks good. But wouldn't it be nice to have the zoom functionality built in to the `Find Next ` and `Find Previous` button clicks as well? This would be handy in large graphs. Maybe a checkbox to enable zooming on all these buttons or similar?

**Review by Delphinus8821 — COMMENTED** [review]
Also thinks it look good and agree with @Auriga2. 
Would it also be good to have this zoom option on the replace and advanced tabs?

**Comment by OrionsGuardian** [comment]
Agree with both @Auriga2 and @Delphinus8821 ...
Zooming should be controlled by a checkbox, and should work with all Find options (Basic and Advanced).

**Review by antares1470 — APPROVED** [review]

**Comment by Auriga2** [comment]
Find next and Find previous buttons don't seem to zoom when the check box is checked while searching in all open graphs. If Find all button is used it zooms in all the graphs, then clicking previous and next buttons will keep that zoom. But after it's zoomed to a specific node, then you change the search text and press find next or previous buttons, it'll still keep the previous zoomed area.

Would it be helpful if we reset view if the check box is unchecked?

**Comment by Delphinus8821** [comment]
> Find next and Find previous buttons don't seem to zoom when the check box is checked while searching in all open graphs. If Find all button is used it zooms in all the graphs, then clicking previous and next buttons will keep that zoom. But after it's zoomed to a specific node, then you change the search text and press find next or previous buttons, it'll still keep the previous zoomed area.
> 
> Would it be helpful if we reset view if the check box is unchecked?

When I use the Find Next and Find Previous buttons when searching across all open graphs it zooms in correctly except for the very first time it swaps to a graph and then it is fine after that. Not sure why it wouldn't be working for you

**Comment by Auriga2** [comment]
Did you try changing the search node? Sometimes it takes a few clicks for `Find All` button as well.
Find Zoom

**Comment by Auriga2** [comment]
It looks better now, but if there's a graph that doesn't contain the search item, the search seem to stuck at a one graph.  
Also the search window hides behind the main window upon clicking any button now.

**Comment by Delphinus8821** [comment]
> It looks better now, but if there's a graph that doesn't contain the search item, the search seem to stuck at a one graph. Also the search window hides behind the main window upon clicking any button now.

The issue where it gets stuck on one result when one of the graphs does not have a result for that search term seems to be a pre-existing issue as it happens on master and in v2.11 so I will make a new ticket for that.

I'm not sure what you mean by the search window is hiding behind the main window, I haven't been able to reproduce that.

**Comment by Auriga2** [comment]
It woks initially. But then selecting to search in all graphs and finding a few items cause the pop up to hide behind the main window.
Find window hide

**Comment by Delphinus8821** [comment]
> It woks initially. But then selecting to search in all graphs and finding a few items cause the pop up to hide behind the main window. Find window hide [ Find window hide ](https://private-user-images.githubusercontent.com/64826928/346430760-3814a8ed-a15b-4a65-8613-ddbbbdac2db5.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjA0MTQwMjIsIm5iZiI6MTcyMDQxMzcyMiwicGF0aCI6Ii82NDgyNjkyOC8zNDY0MzA3NjAtMzgxNGE4ZWQtYTE1Yi00YTY1LTg2MTMtZGRiYmJkYWMyZGI1LmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzA4VDA0NDIwMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWViYmE3Mjg5Y2I1MDc0Nzk1NTA2ZGJlMWZjMTRjOTc5Y2FmZmIyOWM4Y2EwOTcxMjNmYmVmODY1YzNkZGZlOTImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.oO2VQ7gXvO0edOwBgwNwTBVH0RJZb0x5bJ1HXWjyplk) [ ](https://private-user-images.githubusercontent.com/64826928/346430760-3814a8ed-a15b-4a65-8613-ddbbbdac2db5.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjA0MTQwMjIsIm5iZiI6MTcyMDQxMzcyMiwicGF0aCI6Ii82NDgyNjkyOC8zNDY0MzA3NjAtMzgxNGE4ZWQtYTE1Yi00YTY1LTg2MTMtZGRiYmJkYWMyZGI1LmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzA4VDA0NDIwMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWViYmE3Mjg5Y2I1MDc0Nzk1NTA2ZGJlMWZjMTRjOTc5Y2FmZmIyOWM4Y2EwOTcxMjNmYmVmODY1YzNkZGZlOTImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.oO2VQ7gXvO0edOwBgwNwTBVH0RJZb0x5bJ1HXWjyplk)

This happens if you dock the find view window to the main window and then float it again. I have created a new ticket for this as this issue does occur in previous versions (#2121)

**Comment by Auriga2** [comment]
I didn't dock the find view window to the main window, but would have moved it around to trigger the same thing. 

A new issue, can't see the zoom working in the `Basic find` tab and `Current graph` selected on `Advanced find` tab.

**Review by Auriga2 — APPROVED** [review]

**Comment by Auriga2** [comment]
Looks good now. Should we update the help doc images with the new button now?

**Review by Quasar985 — APPROVED** [review]

**Comment by antares1470** [comment]
> Looks good now. Should we update the help doc images with the new button now?

Agreed. That would be good

**Pull request merged**